cpp/benchmarks/stochastic_algorithm.cc

```diff
@@ -0,0 +1,131 @@
+#include "purify/config.h"
+#include "purify/types.h"
+#include <array>
+#include <random>
+#include <benchmark/benchmark.h>
+#include "benchmarks/utilities.h"
+#include "purify/algorithm_factory.h"
+#include "purify/directories.h"
+#include "purify/measurement_operator_factory.h"
+#include "purify/mpi_utilities.h"
+#include "purify/operators.h"
+#include "purify/utilities.h"
+#include "purify/uvw_utilities.h"
+#include "purify/wavelet_operator_factory.h"
+#include <sopt/imaging_padmm.h>
+#include <sopt/mpi/communicator.h>
+#include <sopt/mpi/session.h>
+#include <sopt/power_method.h>
+#include <sopt/relative_variation.h>
+#include <sopt/utilities.h>
+#include <sopt/wavelets.h>
+#include <sopt/wavelets/sara.h>
+
+#ifdef PURIFY_H5
+#include "purify/h5reader.h"
+#endif
+
+using namespace purify;
+
+class StochasticAlgoFixture : public ::benchmark::Fixture {
+ public:
+  void SetUp(const ::benchmark::State &state) {
+    m_imsizex = state.range(0);
+    m_imsizey = state.range(0);
+
+    m_sigma = 0.016820222945913496 * std::sqrt(2);  // see test_parameters file
```

Comment:
We should probably not propagate this comment across the repo since we have no idea what it means and no such file exists